src/firefly/js/metaConvert/DataLinkProcessor.js

```diff
@@ -50,14 +50,17 @@ const isTooBig= (size) => size>GIG;
  * @param {TableModel} params.datalinkTable
  * @param {ActivateParams} params.activateParams
  * @param {Array.<DataProductsDisplayType>} [params.additionalServiceDescMenuList]
+ * @param {string} params.dlUrl datalink url - useful to get dataSource for catalog (non obscore) tables
  * @param {boolean} [params.doFileAnalysis]
  * @param {boolean} [params.onlyRelatedGrid] - if true then the datalink tables much have multiple image grid semantics
  * @param {string} [params.baseTitle]
  * @return {DataProductsDisplayType}
  */
 export function processDatalinkTable({sourceTable, row, datalinkTable, activateParams, baseTitle=undefined,
-                                     additionalServiceDescMenuList, doFileAnalysis=true, onlyRelatedGrid = false}) {
-    const dataSource= getObsCoreAccessURL(sourceTable,row);
+                                     additionalServiceDescMenuList, dlUrl, doFileAnalysis=true, onlyRelatedGrid = false}) {
+    let dataSource= getObsCoreAccessURL(sourceTable,row);
+    //for catalog tables, datalink tables are not linked via access_url, so dataSource will be empty from the above call
+    if (dataSource === '') dataSource = dlUrl;
```

Comment:
@robyww the reason the cutouts and spectra were showing, but the datalink table wasn't coming up is because for our catalog (_MER-catalog-search.xml_) table, the call to `getObsCoreAccessURL` would return an empty string. Because that would attempt to get the access_url links to the datalink tables (which an obscore type table like _MER-tiles-search.xml_ has, but our catalog table doesn't). 

So instead I decided to pass the datalink url directly from `ServDescConverter.js` to the `processDatalinkTable` function, and use it when the dataSource is empty (in the case of non-obscore tables). 

With this fix, the cutouts, spectra and datalink table all show up in the results view (but I'll need to update the test build and push new endpoint changes to kubernetes for this to work for others).